### DISCUSSION TITLE
Default An unphysical results in the droplet impact tutorial

### DISCUSSION THREAD

**Discussion by Ali19881366** [fppimenta/rheoTool#56] [General]
Hi everybody

I ran the droplet impact tutorial. However, when I saw the results in ParaView, I faced some teeth on the droplet perimeter (the interface with the air), as can be seen in the following pic.

https://pasteboard.co/K2qPPzE.jpg

While in the reference paper one doesn't see these teeth.

https://pasteboard.co/K2qPiIi.jpg

I have made many changes in the setup case, e.g. increasing and decreasing Betta, increasing Re, decreasing Wi, decreasing and increasing the dt, generating a very smaller grid, etc. But the teeth remained! 

Even, I set zero values for etaP and lambda, showing a Newtonian fluid, but they remained. However, when I chose Newtonian constitutive eq. the teeth would be removed. This showed me that this is just because of the numerical process of discretization or solution of Oldroyd-B constitutive eq. 

Note: Can you explain to me physically why the authors set zero value for surface tension?
As we know, the droplet impact behaviour is totally under the effect of surface tension forces. Nevertheless, when I set a non-zero value for the surface tension coef, e.g. 0.07 N/m, teeth got smaller. But the reference paper solved this problem using sigma = 0 and I must solve it in this situation and get the correct answer.

Do you know what they are, and how to solve them?

Thanks

**Comment by fppimenta** [comment]
I don't know the changes you have made to the original tutorial nor the code versions you are using, but what I get with the most recent version run in of 7.0 is this:

![ImpactingDrop](https://user-images.githubusercontent.com/22405564/118796854-e5339b80-b893-11eb-9b71-a857baf21664.gif)

There is in fact one major difference to the reference but it is not related with spikes. The cause has been identified and will be fixed in a future release. I will make this warning again: rheoInterFoam is under development, which means it is not ready for final use. Only use it if you know exactly what the solver is doing and agree with that. Otherwise, wait for a final version of this solver.

Problems don't need to be physical to test numerical methods. However, in this particular case having a zero (or very low) surface tension can in fact be physical for Ca -> +Inf, which would be the case of a very viscous fluid.

**Comment by Ali19881366** [comment]
Dear Francisco

Thanks for your reply. I appreciate you and your team's effort in developing this valuable package. 

I didn't make any change within the tutorial problem. I'm using rheoTool-5 (last version), on OF-7, on ubuntu-20.04. I don't know why this problem has been happened in my case, while yours did not have that!

May I know the approximate date of releasing the new version of rheoTool package? And could you please tell me about the approximate date of final version? Will the future version be the final version?

What is strange for me is that when I set zero value for etaP and lambda, and etaS=4 (the same total viscosity of initial problem) showing a Newtonian fluid, the spikes remained. However, when I chose Newtonian constitutive eq. the teeth would be removed. This showed me that this is just because of the numerical process of discretization or solution of Oldroyd-B constitutive eq. This issue has not been solved for me.

Sincerely

**Comment by fppimenta** [comment]
Well, that's even stranger for me, because if you set lambda.water = 0 the solver should simply stop (give an error) on the first time-step, since you get division by 0. Simulating a Newtonian fluid with a VE model is not a good idea. If your case runs with lambda.water = 0 (read it like: lambda exactly equal to zero), then something is definitively wrong with your install. I suggest you to install vbox, create a vmachine and make fresh installs of openfoam 7.0 (binaries will be enough) and rheoTool, following closely the instructions for both. On fresh installs, you should get what I showed above, which, however, does not invalidate what I said about the experimental character of rheoInterFoam.

Unfortunately, I cannot give you a date (not even approximate) for the effective release of rheoInterFoam, as it depends on a lot of factors. It can be one month as well as one year or more. It has been 'under-development' since the initial release of rheoTool in 2016. Until an effective release of the solver, I strongly advise you not to use it unless you know what you are doing.

Note: rheoInterFoam is the only solver of rheoTool which is not ready for use. All the others are ok, i.e. they have been thoroughly tested, although this does not mean that they are bug free (I guess no real code can claim that).

**Reply by Ali19881366** [reply]
Thank you!

I think I wrote sth not very exact. The etaP was set exactly zero and the lambda was set a very small value. This value is small enough to represent the Newtonian behavior and also large enough to evoid divergence. I had tested it before for the flow on a cylinder and compared with Newtonian case and got almost similar results.

Regards
Ali

**Comment by Ali19881366** [comment]
Dear Francisco

I found the reason, but not the solution! I'm sorry, I forgot that I changed the grid. 
In order to see the droplet interface better, I used a smaller grid, while other settings remained constant. I changed the grid from (120 120 1) (1 1 1 )  to  (120 180 1) (5 5 1). In fact, in such a rough grid (initial grid in the tutorial) the interface cannot be seen as well as we need. So, we decided to make the grid smaller and faced those spikes. 

Anyway, If you have any suggestions on how to have both a smaller grid (to see a high-quality interface) and avoiding the creation of spikes, inform me, please.

Thanks

**Comment by jaspinder1691** [comment]
Hi,

I recently installed OpenFOAM and RheoTool. I want to simulate a case involving the buoyancy-induced motion of a Newtonian droplet within an elastoviscoplastic material. Specifically, a Newtonian fluid droplet is introduced from the top of a cylinder filled with an elastoviscoplastic liquid, allowing the droplet to descend through the non-Newtonian material.

Could you guide me on how to approach this simulation? I need help using RheoTool with OpenFOAM 9. I tried starting with the impacting droplet tutorial, but it's not running correctly. Any suggestions on setup or troubleshooting would be greatly appreciated!
